zeus/carbon/carbon.py

```diff
@@ -0,0 +1,95 @@
+"""Carbon intensity providers used for carbon-aware optimizers."""
+
+from __future__ import annotations
+
+import abc
+import json
+import logging
+import requests
+
+logger = logging.getLogger(__name__)
+
+
+def get_ip_lat_long() -> tuple[float, float]:
+    """Retrieve the latitude and longitude of the current IP position."""
+    try:
+        ip_url = "http://ipinfo.io/json"
+        resp = requests.get(ip_url)
+        loc = resp.json()["loc"]
+        lat, long = map(float, loc.split(","))
+        logger.info("Retrieved latitude and longitude: %s, %s", lat, long)
+        return lat, long
+    except requests.exceptions.RequestException as e:
+        logger.exception(
+            "Failed to retrieve current latitude and longitude of IP: %s", e
+        )
+        raise
+
+
+class CarbonIntensityNotFoundError(Exception):
```

Comment:
Defining custom errors is good practice, thanks. Please inherit from `zeus.exception.ZeusBaseError` and prefix the name of the exception with `Zeus`.